modules/programs/antidote.nix

```
@@ -0,0 +1,71 @@
+{ config, lib, pkgs, ... }:
+
+with builtins;
+with lib;
+let
+  module_config = config.programs.zsh.antidote;
+
+  zPluginStr = with lib;
+    (pluginNames:
+      optionalString (pluginNames != [ ]) "${concatStrings (map (name: ''
+        ${name}
+      '') pluginNames)}");
+  trimLines = (text:
+    concatLines (map (p: removePrefix "     " p) (splitString "\n" text)));
+in {
+  options.programs.zsh.antidote = {
+    enable = mkEnableOption "antidote - a zsh plugin manager";
+
+    plugins = mkOption {
+      default = [ ];
+      type = types.listOf types.str;
+      description = "List of antidote plugins, strings.";
+    };
+
+    useFriendlyNames = mkOption {
+      default = false;
+      type = types.bool;
+      description = "use friendlyPath.";
+    };
```

Not sure what this actually does but I think something like this would be simpler:

```suggestion
    useFriendlyNames = mkEnableOption "friendly names";
```